### PR TITLE
PR strange behaviour

### PR DESCRIPTION
**Pull request by nevolin-dmitry-leonid**
test code

```
from typing import Dict

import torch
import torch.utils.data as data_utils

from piq import PR as _PR
from piq.functional.base import crop_patches


class _ComputeFeatsDataset(data_utils.Dataset[Dict[str, torch.Tensor]]):
    def __init__(self, batch: torch.Tensor) -> None:
        self.batch = crop_patches(batch)

    def __getitem__(self, index: int) -> Dict[str, torch.Tensor]:
        return {'images': self.batch[index]}

    def __len__(self) -> int:
        return int(self.batch.shape[0])


def _main() -> None:
    reference = torch.ones(1, 3, 256, 256)
    distorted = reference

    reference_data_loader = data_utils.DataLoader(_ComputeFeatsDataset(reference))
    distorted_data_loader = data_utils.DataLoader(_ComputeFeatsDataset(distorted))

    pr_ = _PR()
    reference_features = pr_.compute_feats(reference_data_loader, device='cpu')
    distorted_features = pr_.compute_feats(distorted_data_loader, device='cpu')
    score = pr_(reference_features, distorted_features)

    precision, recall = float(score[0].numpy()), float(score[1].numpy())

    print(f"{precision=}")
    print(f"{recall=}")


if __name__ == '__main__':
    _main()

```

prints

```
precision=0.0
recall=0.9795918464660645
```

but they seems to be

```
precision=1.0
recall=1.0
```
i found potentially mismatching with paper, original formula

image

actual formula

```
        precision = (
                distance_real_fake < real_nearest_neighbour_distances.unsqueeze(1)
        ).any(dim=0).float().mean()

        recall = (
                distance_real_fake < fake_nearest_neighbour_distances.unsqueeze(0)
        ).any(dim=1).float().mean()
```

so, **<** seems to be **<=**

commit in pull request is checking **<=** with precision (due to float errors)

**PS:** (offtop) in paper they define additional metric "Realism Score" - are there plans to add it? Is that even necessary? I may try to add Realism Score as another pull request later.

### CODE REVIEW

**Review by zakajd — CHANGES_REQUESTED** [review]
Thanks for noticing this bug!
Can you also add test into `tests/test_pr.py` to check for correctness in CI run?

**Comment by zakajd** [comment]
> in paper they define additional metric "Realism Score" - are there plans to add it? Is that even necessary? I may try to add Realism Score as another pull request later.

We didn't plan to add Realism Score, but if you wish to open a PR it's always welcomed 👌🏼

**Review by zakajd — APPROVED** [review]
LGTM 👍🏼

**Comment by snk4tr** [comment]
I will have a look ASAP.
